benchmarks/bench_evolve.py

```diff
@@ -46,7 +46,7 @@ def benchmark_write_operator_card_from_file_num_fonll(
         with open(opcard_path, encoding="utf-8") as f:
             ocard = yaml.safe_load(f)
         for entry in ocard["mugrid"]:
-            assert entry[1] == nfff
+            assert entry[1] == int(nfff)
```

Comment:
I think this should be saved/read as `int` so no need to force the int here

Reply:
I believe it is needed because here `nfff` comes from the `numpy` array of `MIXED_FNS_CONFIG` and thus is a string. I tried without and the benchmark was not passing

Reply:
Oh wait you get it from the hardcoded list, but then it should already be coded as int right? Meaning this does nothing

Reply:
I think the benchmark wasn't passing because in the tcard it was saved as a string, but you were comparing it to MIXED_FNS_CONFIG where it is an int

Reply:
See [bc78002](https://github.com/NNPDF/pineko/pull/99/commits/bc7800238693498afb41963025a5550e1c21485e)

Reply:
Yes but look at [b284bb1](https://github.com/NNPDF/pineko/pull/99/commits/b284bb1f03b8751dd18803b2b2ca7eec61b9bb5d) where another benchmark is failing for the same reason. You see that the string is actually the `'3'` coming from MIXED_FNS_CONFIG

Reply:
It is not coming from the theory card

Reply:
Does it help if you provide the `dtypes` for each of the components in the tuple?

Reply:
I don't believe so. I believe that the problem is linked to the `np.array` cast

Reply:
Yes and I believe that can be changed for the individual components of the tuples by doing `MIXED_FNS_CONFIG = np.array(dtype="U10,int,U10"`

Reply:
If I do like you suggested then the array shape becomes `(7,)` instead of `(7,3)` as it should and the transposition does not work anymore